MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -2999,39 +3536,38 @@ public void changeEdgeTrigger(String name) {
     }
 
     /**
-     * This function returns an html-coded tooltip that says what edge will be used
      *
-     * @return
+     * @return an html-coded tooltip that says what edge will be used
      */
     public String getEdgeTooltip() {
-        String edgett = "";
+        StringBuilder edgett = new StringBuilder();
         for (Enumeration<IOption> i = getOptions(PilotOptions.EDGE_ADVANTAGES); i.hasMoreElements(); ) {
             IOption ability = i.nextElement();
             //yuck, it would be nice to have a more fool-proof way of identifying edge triggers
             if (ability.getName().contains("edge_when") && ability.booleanValue()) {
-                edgett = edgett + ability.getDescription() + "<br>";
+                edgett.append(ability.getDescription()).append("<br>");
             }
         }
-        if (edgett.equals("")) {
+        if (edgett.toString().equals("")) {
             return "No triggers set";
         }
         return "<html>" + edgett + "</html>";
     }
+    //endregion edge
 
     /**
-     * This function returns an html-coded list that says what abilities are enabled for this pilot
      *
-     * @return
+     * @return an html-coded list that says what abilities are enabled for this pilot
      */
     public String getAbilityList(String type) {
-        String abilityString = "";
+        StringBuilder abilityString = new StringBuilder();
         for (Enumeration<IOption> i = getOptions(type); i.hasMoreElements(); ) {
             IOption ability = i.nextElement();
             if (ability.booleanValue()) {
-                abilityString = abilityString + Utilities.getOptionDisplayName(ability) + "<br>";
+                abilityString.append(Utilities.getOptionDisplayName(ability)).append("<br>");
             }
         }
-        if (abilityString.equals("")) {
+        if (abilityString.toString().equals("")) {
```

Comment:
Is `abilityString.length() == 0` not an option? (or `size() == 0`) ?

Reply:
It is, swapped over to using abilityString.length() == 0